.github/workflows/tests.yml

```diff
@@ -0,0 +1,128 @@
+name: Tests
+
+on:
+
+  # On which repository actions to trigger the build.
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allow job to be triggered manually.
+  workflow_dispatch:
+
+# Cancel in-progress jobs when pushing to the same branch.
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+
+  tests:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+
+      # Run all jobs to completion (false), or cancel
+      # all jobs once the first one fails (true).
+      fail-fast: false
```

Comment:
I think a good default is to use `fail-fast: true`. I usually only disable failing fast, if I want to get the **whole picture at once** about all failing tests on specific environments.

In daily business, it will save a bunch of resources when the jobs on the other environments get canceled, if one environment fails, because this is what you will usually want in this case.

Do you agree?

Reply:
I don't see harm in the idea

Reply:
Thanks, I've updated the patch accordingly.